### PR TITLE
chore: add error boundaries

### PR DESCRIPTION
**Pull request by seaerchin**
## Problem
We are missing error boundaries for common errors thrown from our backend. This results in a degraded user experience because our user then sees an ugly error page

## Solution
1. add default error boundaries for the common error cases

## Screenshots
see provided storybooks

### CODE REVIEW

File: apps/studio/src/components/ErrorBoundary/DefaultServerError.tsx
Comment (dcshzj) [inline apps/studio/src/components/ErrorBoundary/DefaultServerError.tsx:21]: nit: spacing

```suggestion
          It's not you, it's us. Please try refreshing this page.
```

File: apps/studio/src/components/ErrorBoundary/DefaultServerError.tsx
Comment (dcshzj) [inline apps/studio/src/components/ErrorBoundary/DefaultServerError.tsx:25]: nit: support

```suggestion
            contact Isomer Support
```

**Review by dcshzj — APPROVED** [review]

File: apps/studio/src/components/ErrorBoundary/DefaultServerError.tsx
Comment (sehyunidaaa) [reply apps/studio/src/components/ErrorBoundary/DefaultServerError.tsx:25]: @seaerchin this one is mb - wasn't aware we were using Support throughout. i'll update Figma too

**Comment by seaerchin** [comment]
i'll merge once @sehyunidaaa approves

**Pull request merged**
